server/handlers.go

```diff
@@ -52,80 +56,502 @@ func handleRequestDownloaderMissing(logger *zap.Logger) http.HandlerFunc {
+		})
+
+		if err != nil {
+			logger.Error("failed to draw ring for account identity", zap.Error(err))
 			return
 		}
-		mime, err := images.GetProtobufImageMime(payload)
+	}
+
+	if parsed.IndicatorSize != 0 {
+		payload, err = images.AddStatusIndicatorToImage(payload, parsed.IndicatorColor, parsed.IndicatorSize, parsed.IndicatorBorder)
 		if err != nil {
-			logger.Error("failed to get mime", zap.Error(err))
+			logger.Error("failed to draw status-indicator for initials", zap.Error(err))
+			return
 		}
+	}
 
-		w.Header().Set("Content-Type", mime)
-		w.Header().Set("Cache-Control", "no-store")
+	if len(payload) == 0 {
+		logger.Error("handleAccountInitialsImpl: empty image", zap.String("keyUid", parsed.KeyUID), zap.Error(err))
+		return
+	}
+	mime, err := images.GetProtobufImageMime(payload)
+	if err != nil {
+		logger.Error("failed to get mime", zap.Error(err))
+	}
 
-		_, err = w.Write(payload)
+	w.Header().Set("Content-Type", mime)
+	w.Header().Set("Cache-Control", "no-store")
+
+	_, err = w.Write(payload)
+	if err != nil {
+		logger.Error("failed to write image", zap.Error(err))
+	}
+}
+
+func handleAccountInitialsPlaceholder(logger *zap.Logger, w http.ResponseWriter, r *http.Request, parsed ParsedParams) {
+	if parsed.FullName == "" {
+		logger.Error("handleAccountInitialsPlaceholder: no full name")
+		return
+	}
+
+	initials := images.ExtractInitials(parsed.FullName, parsed.InitialsLength)
+
+	payload, err := images.GenerateInitialsImage(initials, parsed.BgColor, parsed.Color, parsed.FontFile, parsed.BgSize, parsed.FontSize, parsed.UppercaseRatio)
+
+	if err != nil {
+		logger.Error("handleAccountInitialsPlaceholder: failed to generate initials image.", zap.String("keyUid", parsed.KeyUID), zap.String("name", parsed.FullName), zap.Error(err))
+		return
+	}
+
+	if parsed.IndicatorSize != 0 {
+		payload, err = images.AddStatusIndicatorToImage(payload, parsed.IndicatorColor, parsed.IndicatorSize, parsed.IndicatorBorder)
 		if err != nil {
-			logger.Error("failed to write image", zap.Error(err))
+			logger.Error("failed to draw status-indicator for initials", zap.Error(err))
+			return
+		}
+	}
+
+	if len(payload) == 0 {
+		logger.Error("handleAccountInitialsPlaceholder: empty image", zap.String("keyUid", parsed.KeyUID), zap.Error(err))
+		return
+	}
+	mime, err := images.GetProtobufImageMime(payload)
+	if err != nil {
+		logger.Error("failed to get mime", zap.Error(err))
+	}
+
+	w.Header().Set("Content-Type", mime)
+	w.Header().Set("Cache-Control", "no-store")
+
+	_, err = w.Write(payload)
+	if err != nil {
+		logger.Error("failed to write image", zap.Error(err))
+	}
+}
+
+func handleAccountInitials(multiaccountsDB *multiaccounts.Database, logger *zap.Logger) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		params := r.URL.Query()
+		parsed := ParseParams(logger, params)
+
+		if parsed.FontFile == "" {
```

Comment:
I'm curious how we are passing a FontFile from the client to status-go?
Are we passing the path to it?

Reply:
Yes, the absolute path to the font file. mobile use react-native-fs to get the path.

Reply:
Thanks for the clarification.

Reply:
mobile code at https://github.com/status-im/status-mobile/pull/16193/files#diff-ddfc24fc1622b950e08d907033895039aef6924760bf4e7670dce02dea1c6aeaR16